pkg/scheduler/plugins/reservation/nominator.go

```diff
@@ -149,6 +216,47 @@ func (pl *Plugin) GetNominatedReservation(pod *corev1.Pod, nodeName string) *fra
 	return pl.reservationCache.getReservationInfoByUID(reservationID)
 }
 
+func (pl *Plugin) BeforeFilter(ctx context.Context, cycleState *framework.CycleState, pod *corev1.Pod, nodeInfo *framework.NodeInfo) (*corev1.Pod, *framework.NodeInfo, *framework.CycleState, bool, *framework.Status) {
+	if !reservationutil.IsReservePod(pod) {
+		return pod, nodeInfo, cycleState, false, nil
+	}
+
+	nominatedReservationInfos := pl.nominator.NominatedReservationForNode(nodeInfo.Node().Name)
+	if len(nominatedReservationInfos) == 0 {
+		return pod, nodeInfo, cycleState, false, nil
+	}
+
+	if nodeInfo.Node() == nil {
+		// This may happen only in tests.
+		return pod, nodeInfo, cycleState, false, nil
+	}
+
+	nodeInfoOut := nodeInfo.Clone()
+	stateOut := cycleState.Clone()
+
+	rName := reservationutil.GetReservationNameFromReservePod(pod)
+	_, err := pl.rLister.Get(rName)
+	if err != nil {
+		return pod, nodeInfo, cycleState, false, framework.NewStatus(framework.Error, "reservation not found")
+	}
+
+	for _, rInfo := range nominatedReservationInfos {
+		if schedulingcorev1.PodPriority(rInfo.Pod) >= schedulingcorev1.PodPriority(pod) && rInfo.Pod.UID != pod.UID {
+			pInfo := framework.NewPodInfo(rInfo.Pod)
+			nodeInfoOut.AddPodInfo(pInfo)
+			status := pl.handle.RunPreFilterExtensionAddPod(ctx, stateOut, pod, pInfo, nodeInfoOut)
```

Comment:
IMO it's enough to update the existing cycleState here, so we don't need to clone, let alone return a new cycleState.  WDYT?

Reply:
done